tools/FakeItEasy.Build/Program.cs

```diff
@@ -0,0 +1,157 @@
+        private const string VersionInfoFile = "./src/VersionInfo.cs";
+        private const string RepoUrl = "https://github.com/FakeItEasy/FakeItEasy";
+
+        private static readonly string[] ProjectsToPack =
+        {
+            "./src/FakeItEasy/FakeItEasy.csproj",
+            "./src/FakeItEasy.Analyzer/FakeItEasy.Analyzer.CSharp.csproj",
+            "./src/FakeItEasy.Analyzer/FakeItEasy.Analyzer.VisualBasic.csproj"
+        };
+
+        private const string AnalyzerMetaPackageNuspecPath = "./src/FakeItEasy.Analyzer.nuspec";
+
+        private static readonly string[] Pdbs =
+        {
+            "src/FakeItEasy/bin/Release/net40/FakeItEasy.pdb",
+            "src/FakeItEasy/bin/Release/netstandard1.6/FakeItEasy.pdb",
+            "src/FakeItEasy.Analyzer/bin/Release/FakeItEasy.Analyzer.Csharp.pdb",
+            "src/FakeItEasy.Analyzer/bin/Release/FakeItEasy.Analyzer.VisualBasic.pdb"
+        };
+
+        private static readonly IReadOnlyDictionary<string, string[]> TestSuites = new Dictionary<string, string[]>
+        {
+            ["unit"] = new[]
+            {
+                "tests/FakeItEasy.Tests",
+                "tests/FakeItEasy.Analyzer.CSharp.Tests",
+                "tests/FakeItEasy.Analyzer.VisualBasic.Tests",
+            },
+            ["integ"] = new[]
+            {
+                "tests/FakeItEasy.IntegrationTests",
+                "tests/FakeItEasy.IntegrationTests.VB",
+            },
+            ["spec"] = new[]
+            {
+                "tests/FakeItEasy.Specs"
+            },
+            ["approve"] = new[]
+            {
+                "tests/FakeItEasy.Tests.Approval"
+            }
+        };
+
+        private static string LogsDirectory = "./artifacts/logs";
+        private static readonly string OutputDirectory = Path.GetFullPath("./artifacts/output");
+
+        private static readonly string MsBuild = $"{GetVSLocation()}/MSBuild/15.0/Bin/MSBuild.exe";
+
+        static void Main(string[] args)
+        {
+            Target("default", DependsOn("unit", "integ", "spec", "approve", "pack"));
+
+            Target("outputDirectory", () => Directory.CreateDirectory(OutputDirectory));
+
+            Target("logsDirectory", () => Directory.CreateDirectory(LogsDirectory));
+
+            Target("testsDirectory", () => Directory.CreateDirectory(TestsDirectory));
+
+            Target("build", DependsOn("clean", "restore", "versionInfoFile"), () => RunMsBuild("Build"));
+
+            Target("versionInfoFile", () => Run(ToolPaths.GitVersion, $"/updateAssemblyInfo {VersionInfoFile} /ensureAssemblyInfo"));
+
+            Target("clean", DependsOn("logsDirectory"), () => RunMsBuild("Clean"));
+
+            Target(
+                "restore",
+                () => Run("dotnet", $"restore"));
+
+            Target(
+                "unit",
+                DependsOn("build", "testsDirectory"),
+                () => RunTests("unit"));

```

Comment:
A bonus change, for another review, but I think we could use the fancy new "target that iterates" feature on these tests.

Reply:
Of course. I created #1416.